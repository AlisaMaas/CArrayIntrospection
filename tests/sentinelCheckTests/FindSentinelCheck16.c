/**
* This check tests we detect no sentinel checks in a loop with
* a "backwards" sentinel check (breaking if not null)
*
* We expect to find no sentinel check.
**/
void print(char* string){}
int find(char string[], int distance)
{
  for(int i = 0; i < distance; i++){
		if(string[i] != '\0'){
			break;
		}
	}
  return 1;
}
