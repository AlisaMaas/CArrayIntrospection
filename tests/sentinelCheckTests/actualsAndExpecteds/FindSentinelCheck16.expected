Printing analysis 'Promote Memory to Register' for function 'print':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'find':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Find each branch used to exit a loop when a sentinel value is found in an array' for function 'print':
Analyzing function: print
	We found: 0 loops
Printing analysis 'Find each branch used to exit a loop when a sentinel value is found in an array' for function 'find':
Analyzing function: find
	We found: 1 loops
	There are 0 sentinel checks in this loop(for.cond)
	We can bypass all sentinel checks.
	Sentinel checks: 
